docs/grammar.md

```diff
@@ -0,0 +1,118 @@
+# RON grammar
+
+This file describes the structure of a RON file in [EBNF notation][ebnf].
+If extensions are enabled, some rules will be replaced. For that, see the
+[extensions document][exts] which describes all extensions and what they override.
+
+[ebnf]: https://en.wikipedia.org/wiki/Extended_Backus–Naur_form
+[exts]: ./extensions.md
+
+## RON file
+
+```ebnf
+RON = [extensions], ws, value, ws
+```
+
+## Whitespace and comments
+
+```ebnf
+ws = { ws_single, comment }
+ws_single = "\n" | "\t" | "\r" | " "
+comment = ["//", { no_newline }, "\n"]
+```
+
+## Commas
+
+```ebnf
+comma = ws, ",", ws
+```
+
+## Extensions
+
+```ebnf
+extensions = { "#", ws, "!", ws, "[", ws, extensions_inner, ws, "]", ws }
```

Comment:
I don't see any reason to support `ws` between `#` and `!`

Reply:
That's what Rust allows, too. And that's what I implemented. Should I change the implementation?